Documentation and checklist updates for onboarding.

adunkman opened the pull request:
Draws upon work from existing documentation, including the [U.S. Tax Court onboarding document](https://docs.google.com/document/d/1ktys8x_XcFGCto1wxL5uUeAZlk87lOr31FRrZOuL_8g/edit) and Flexion’s [Onboarding Quick Reference Guide](https://docs.google.com/document/d/1jOd0wAsqTDBKsFeEvUj9ezvdTimTsqTwkYd6Imibl3c/edit#heading=h.jrlttmsqvst3) and [Team Working Agreement](https://docs.google.com/document/d/1AFBrxBBJD7CZoO7Go6hfTHCqqWLi00Eok9mGaCX3VGo/edit), this pull request updates the repository README, How We Work page, and adds two new issue templates for onboarding and offboarding employees. 

<details><summary>When reviewing, I suggest toggling views to "Display the rich diff" which will render the documents.</summary>

![Screenshot showing rich diff toggle](https://user-images.githubusercontent.com/14930/120829843-df90b380-c52b-11eb-8329-a1fea3604a02.png)

</details>

adunkman committed:
Update README.

adunkman committed:
Add note about Flexion and 18F to overview.

adunkman committed:
Update backlogs

Drop the migration backlog; add the ATO backlog (still private).

adunkman committed:
Add link to load testing.

adunkman committed:
Add routine monitoring and maintenance checklist

adunkman committed:
Update how we work with how we work. 😉

adunkman committed:
Add user types to README

adunkman committed:
Clarify 18F’s role

adunkman committed:
Add onboarding and offboarding checklists.

adunkman commented:
Tagging Mike and Jessica for review — we can chat this through in-person as well if preferable. 👯‍♀️

kkoskelin commented on .github/ISSUE_TEMPLATE/onboarding.md:
```diff
@@ -0,0 +1,32 @@
+
+- [ ] GitHub — speak with your team lead to determine the appropriate repositories and levels of access in the [ustaxcourt](https://github.com/ustaxcourt) and [flexion](https://github.com/flexion) organizations.
+- [ ] [Slack](https://ustaxcourt.slack.com) — look for important channels to join in [How We Work](./how-we-work.md).
+- [ ] [Sonarcloud](https://sonarcloud.io/organizations/flexion-github/projects)
```
The team's reliance on Sonarcloud has been replaced with Deepscan.

adunkman replied:
Exciting! I see [Sonarcloud is still referenced quite a bit](https://github.com/ustaxcourt/ef-cms/search?q=Sonarcloud) — has it been fully deprecated and removed, or are we still in the process of replacing it? 

I’m happy to remove all references to Sonarcloud, if we’re not using it anymore.

kkoskelin replied:
Amendment: the removal of Sonarcloud will be completed after Batch 1 is merged.

kkoskelin commented on README.md:
```diff
@@ -1,257 +1,116 @@
+- [Documentation](./docs/README.md) about the CI/CD setup, API, style guide, UX, code review, etc.
+- [Ongoing development documentation](./wikiwiki/README.md) such as designs, research data, user workflows, etc.
+- [Product roadmap](https://docs.google.com/document/d/1g3D1zPNQqVsWhJ6uGIVcHA04IdBHgYcOCm9roaSfFNk/edit) 🔒
+- [Goals of DAWSON](https://docs.google.com/document/d/1xCh1hbXjJItOWlQ2MH-GDDc0KWho88KKcHfUJcpHkSs/edit) 🔒
```
This document (Goals) is not accessible to the general public, afaict.

adunkman replied:
Correct! They’re referenced here with a lock icon (🔒) to indicate as such — I think keeping the documentation together and linking out to private sections helps keep everyone on the same page. One example of this pattern is the [TTS Handbook](https://handbook.tts.gsa.gov/) which also adds lock icons to indicate when items are private.

kkoskelin commented on README.md:
```diff
@@ -1,257 +1,116 @@
+| Backlog | Location
+|---------|----------
+| User stories and feature work <br>(Primary backlog) | [GitHub issues in Flexion’s repository](https://github.com/flexion/ef-cms/issues#workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/board?repos=152320868); scrum board visible using the [ZenHub browser plugin](https://www.zenhub.com/extension).
+| Refactoring and bugs | [Trello cards in Flexion’s OpEx/DevEx board](https://trello.com/b/9tgrIFfA/ef-cms-opex-devex). 🔒
```
This trello board is also not visible to the general public from what I can see.

JessicaMarine commented on docs/how-we-work.md:
```diff
@@ -1,97 +1,100 @@
+6.  Pull request merged to staging branch and deployed to the staging environment by the Court
+7.  User testing is performed
+8.  Court creates pull request to production branch and merges to update the production environment
 
```
We might want to outline the two different Git flows: bugs vs. story work.

mmarcotte replied:
For this, Pull requests of story work is a PR directly to the `staging` branch after testing has been conducted in the `migration` environment.

For Bugfixes, Pull requests are opened to the `test` environment from work based off of `staging`. If approved, a new PR is made to the `staging` branch and readied for production deploy.

JessicaMarine commented on docs/how-we-work.md:
```diff
@@ -1,97 +1,100 @@
-8.  **Deliver incremental value.** We deliver value to users as often as possible.
+- **Daily design sync** — this meeting is a time to solution functionality, demo in-progress work with the team and review new wireframes
+- **Daily standup**
 
```
Might want to add something about our Monday & Friday DAWSON touch-base with Judge Buch and Stephanie.

mmarcotte replied:
... a stake holder meeting

### Weekly

- **Stakeholder Meeting**  - on Mondays and Fridays, we review the current project status with stakeholders at the Court and discuss the schedule of large releases that need off-hours deployments.

adunkman committed:
PR Feedback

- Remove sonarcloud
- Add description of bugs and story work flows
- add stakeholder meeting description

adunkman committed:
Merge remote-tracking branch 'origin/staging' into onboarding-documentation-updates

adunkman commented:
@mmarcotte PR updated!
